mypy/checkexpr.py

```diff
@@ -1543,8 +1544,11 @@ def visit_cast_expr(self, expr: CastExpr) -> Type:
         """Type check a cast expression."""
         source_type = self.accept(expr.expr, type_context=AnyType(), allow_none_return=True)
         target_type = expr.type
-        if self.chk.options.warn_redundant_casts and is_same_type(source_type, target_type):
+        options = self.chk.options
+        if options.warn_redundant_casts and is_same_type(source_type, target_type):
             self.msg.redundant_cast(target_type, expr)
+        if options.disallow_implicit_any_types and has_any_from_silent_import(target_type):
+            self.msg.implicit_any_from_silent_import("Target type of cast", target_type, expr)
```

Comment:
Oh, I see what you're saying. Maybe it should say stuff like `Return type is Any instead of Unchecked` (not exact wording). Is that what you mean?

I am actually not sure about this. In another case we are typechecking a function definition (where the user types in `def foo ...`), so there is an error specific to return type and arguments. However, when the user types in Callable, I think talking about Callable as it's own type makes sense. Do you have a strong opinion on this?